### PR TITLE
Default dtype of pandas.Series is now a categorical

### LINKED ISSUE #125: Hypnogram.hypno as a pd.Categorical?
@remrama , related to our discussion about the output of `yasa.Hypnogram.upsample_to_data` in #124, I was thinking that we could in fact set the default dtype of `hyp.hypno` as a [pandas.Categorical](https://pandas.pydata.org/pandas-docs/version/1.4/reference/api/pandas.Categorical.html). This reduces the memory size of an upsampled hypnogram (5 stages, 8 hour TIB, freq="10ms") from 22 MB (dtype=str or dtype=np.int64) to 2.7 MB.

```python
pd.Categorical(hyp.hypno, categories=hyp.labels, ordered=False)
```

### PR DESCRIPTION
**Pull request by raphaelvallat**
Closes #125 

This PR:

- changes the default dtype of `hyp.hypno` as a [pandas.CategoricalDtype](https://pandas.pydata.org/docs/reference/api/pandas.CategoricalDtype.html)
- set the default return dtype from np.int64 to np.int16 in `Hypnogram.as_int()`
- fixes a few typos and improves docstrings

Hopefully this should be pretty straightforward to review. Let me know if you have any questions! We're so close :)

### CODE REVIEW

File: yasa/hypno.py [hunk 1]
```diff
@@ -520,6 +528,16 @@ def consolidate_stages(self, new_n_stages):
             scorer=self.scorer,
         )
 
+    def copy(self):
+        """Return a new copy of the current Hypnogram."""
+        return type(self)(
```
Comment (remrama) [inline yasa/hypno.py:533]: Feel free to replace `type(self)` with `Hypnogram`

File: yasa/hypno.py [hunk 2]
```diff
@@ -520,6 +528,16 @@ def consolidate_stages(self, new_n_stages):
...
+            values=self.hypno.to_numpy(),
```
Comment (remrama) [inline yasa/hypno.py:534]: Feel free to ensure this is a copy.

File: yasa/hypno.py [hunk 3]
```diff
@@ -625,8 +643,8 @@ def plot_hypnogram(self, **kwargs):
         --------
         .. plot::
 
-            >>> from yasa import simulate_hypnogram
-            >>> ax = simulate_hypno(tib=480, seed=88).plot_hypnogram(highlight="REM"")
+            >>> from yasa import simulate_hypno
+            >>> ax = simulate_hypno(tib=480, seed=88).plot_hypnogram(highlight="REM")
```
Comment (remrama) [inline yasa/hypno.py:647]: Nice catch!

btw -- I was going to ask anyways -- should this function be renamed as `simulate_hypnogram`? Just trying to think of consistency across functions (eg `plot_hypnogram`). Not a big deal but this is basically a brand new function so it seems like now would be the time to rename it if you want.

File: yasa/hypno.py [hunk 4]
```diff
@@ -238,6 +241,10 @@ def __init__(self, values, n_stages=5, *, freq="30s", start=None, scorer=None):
         map_accepted = {"S": "SLEEP", "W": "WAKE", "R": "REM"}
         hypno = hypno.replace(map_accepted)
         labels = pd.Series(accepted).replace(map_accepted).unique().tolist()
+        # Change dtype of series to "categorical" (reduces memory)
+        cat_dtype = CategoricalDtype(labels, ordered=False)
```
[the remark this answers is not in the document]
Comment (remrama) [reply yasa/hypno.py:245]: If you decide to include this, probably add a case to unittests where a Hypnogram instance is created with not all available stages and then check it has all _available_ categories.

**Review by remrama — APPROVED** [review]
Sweeeeet. Approve 🚀 

This was a great idea I'm so glad you considered this before release.

I also added some comments that address the last few that were still hanging on #124 when it was merged.

File: yasa/hypno.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (raphaelvallat) [reply yasa/hypno.py:245]: The downside if we use only the available categories is that users will get an error if they then try to add a new category, like an "Artefact" or "Unscored". By using all the stages, we ensure that `hyp.hypno.loc[0] = "Art"` even if there were no artefacts epoch previously in the hypno

File: yasa/hypno.py (hunk 3 above)
Comment (raphaelvallat) [reply yasa/hypno.py:647]: Actually you're right, `simulate_hypnogram` would be better

File: yasa/hypno.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (remrama) [reply yasa/hypno.py:245]: Wouldn't "ART" and "UNS" be included in the available categories though? Or am I still missing something?

[the remark this answers is not in the document]
Comment (raphaelvallat) [reply yasa/hypno.py:245]: Sorry I think I misunderstood 🤦 Are you proposing to include all possible stages (WAKE, NREM, REM, LIGHT, DEEP, N1, etc) regardless of `n_stages`?

Just to be clear, currently for a 2 stages hypnogram the CategoricalDtype will be initialized with 4-categories: WAKE, SLEEP, ART, UNS

File: yasa/hypno.py (hunk 3 above)
Comment (raphaelvallat) [reply yasa/hypno.py]: Done in latest commit!

File: yasa/hypno.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (raphaelvallat) [reply yasa/hypno.py:245]: So I think the current behavior is actually what you are suggesting 🤔

[the remark this answers is not in the document]
Comment (remrama) [reply yasa/hypno.py:245]: So sorry @raphaelvallat you're right. Current behavior is what I'm suggesting. I didn't think it through enough and was only looking at `map_accepted` without considering values present in `accepted`.

Sorry for the confusion, ship it! 🚢

**Pull request merged**